### DISCUSSION TITLE
Automatic size detection based on overdump

### DISCUSSION THREAD

**Discussion by mariomadproductions** [sanni/cartreader#710] [Ideas & Suggestions]
I was thinking it would be useful if the cart reader automatically worked out the real extent of the ROM by purposely overdumping, and then analysing the overdump data. I think with some carts/ROM chips, there are patterns you can pick up that indicate that certain blocks are not real data. This would be useful for carts where the size is not written on the ROM chip, and just in general so that the user can be more certain the correct size has been dumped.

**Comment by sanni** [comment]
The OSCR validates* every dump via a CRC32 check and then renames* the ROM according to no-intro at which point the user knows it has been dumped correctly as bad dumps don't get renamed.

*not implemented yet for the Atari 2600, Emerson Arcadia 2001, Fairchild Channel F, Magnavox Odyssey 2 modules

**Reply by mariomadproductions** [reply]
Ah well I'm asking about this for No-Intro, as most ROMs in the database don't have verified status yet (i.e. two trusted dumps that match). Plus ROMs can have variations that are not documented yet.

**Comment by sanni** [comment]
Well if you overdump a cartridge the read data just loops back to the beginning of the maskrom. So you can detect it easily for cartridges with just one maskrom chip. 
The [PCE module](https://github.com/sanni/cartreader/blob/master/Cart_Reader/PCE.ino#L329) has this approach implemented instead of using the database for getting the ROM size.

**Reply by mariomadproductions** [reply]
Yeah, although some ROMs do actually have the data twice, or instead of repeating, the data is corrupted in subtle ways.

**Reply by mariomadproductions** [reply]
And with different dump tools/configurations, you can get different data in the out of bounds area. So I was thinking that the OSCR could emulate that, to detect the ROM boundary in some cases?
